docs/architecture/adr-002-light-client-implementation.md

```diff
@@ -0,0 +1,784 @@
+    here: we are assuming that rollup operators are honest, which ensures that
+    the genesis state roots remain unique. However, there's a potential risk
+    that a maliciously modified rollup could tamper with this value by altering
+    its software. This is why we also need the `code_commitment` field as well.
+
+3. **code_commitment**
+
+    The `code_commitment` is generated as the output of the rollup zk-circuit
+    compilation. This means that different execution logics or, in a broader
+    sense, different versions of rollups should result in distinct code
+    commitments. Alongside the `genesis_state_root`, the `code_commitment`
+    serves as the second parameter that helps link an incoming aggregated proof
+    to a specific rollup, ensuring uniqueness and integrity. Similarly, this
+    field should also be initialized during client creation, and is only
+    permitted to change with a client upgrade. And not during recoveries.
+
+4. **trusting_period**
+
+    Initially, the `trusting_period` was solely under the `da_params` of the
+    `ClientState`, but it has been elevated to a rollup-wide necessity. This
+    shift is motivated by potential emergency scenarios that might necessitate a
+    rollup halt, affecting clients on counterpart chains as well.
+
+    Given that the client verifies both the core header of the DA and the
+    aggregated proof simultaneously during each update, having separate trusting
+    periods for each doesn't align logically. Instead, when configuring this
+    period, we must account for both the DA layer and the rollup, ensuring that
+    the client's trusting duration is maximum acceptable amount. If, for
+    instance, the rollup requires a shorter trusting window, it dictates when
+    the client expires. This parameter is determined by relayers/users and can
+    be updated during both client recoveries and upgrades.
+
+5. **frozen_height**
+
+    Currently, the Sovereign light client can only be frozen due to misbehavior
+    in the DA layer. It's important to note that misbehavior, in the context of
+    the rollup, doesn't have the same implications. Here, misbehavior refers to
+    a scenario where the passed aggregated proof is initially validated by the
+    verifier but is later discovered to be incorrect. Such a case essentially
+    invalidates the credibility of the integrated ZK verifier into the light
+    client.
+
+    At present, we haven't identified any specific potential misbehavior.
+    However, maintaining this field as a rollup-wide parameter allows for future
+    investigations that may uncover potential scenarios needing attention.
+    Additionally, different rollup nodes might benefit from customized
+    executions, making it more sensible to have this parameter rollup-wide.
+
+    A small note regarding this field: When the client is frozen, `ibc-rs` sets
+    the height to `Height::new(0, 1)`, following the same logic as in `ibc-go`.
```

Comment:
Is a `Height { 0, 1 }` reserved only for frozen heights?

Reply:
Yes. Basically, it acts as flag. In `ibc-go`, the convention is `Height { 0, 0 }` for active and `Height { 0, 1 }` for frozen.

In `ibc-rs`, we don't allow `Height` with `revision_height` zero. So we have `None` and `Some(Height{ 0, 1 })`.